src/utils.js

```diff
@@ -333,19 +325,208 @@ function mapSlideCoordToPixelCoord(options) {
   const affine = buildInverseTransform({
     orientation: options.orientation,
     offset: options.offset,
-    spacing: options.spacing
-  })
+    spacing: options.spacing,
+  });
+
+  return applyInverseTransform({ coordinate: point, affine: affine });
+}
+
+/**
+ * Get coordinate with offset
+ *
+ * @param {object} feature feature
+ * @param {number} offset offset
+ * @returns {array} coordinates with offset
+ */
+function coordinateWithOffset(feature, offset = 70) {
+  const geometry = feature.getGeometry();
+  const coordinates = geometry.getLastCoordinate();
+  const [x, y] = coordinates;
+  return !feature.get(Enums.InternalProperties.Marker) &&
+    feature.get(Enums.InternalProperties.Markup) === Enums.Markup.TextEvaluation
+    ? coordinates
+    : [x - offset, y - offset];
+}
+
+/**
+ * Get view unit suffix
+ *
+ * @param {object} view Map view
+ * @returns {string} unit suffix
+ */
+function getUnitSuffix(view) {
+  const UnitsEnum = { METERS: "m" };
+  const DEFAULT_DPI = 25.4 / 0.28;
+
+  const center = view.getCenter();
+  const projection = view.getProjection();
+  const resolution = view.getResolution();
+
+  const pointResolutionUnits = UnitsEnum.METERS;
+
+  let pointResolution = getPointResolution(
+    projection,
+    resolution,
+    center,
+    pointResolutionUnits
+  );
+
+  const DEFAULT_MIN_WIDTH = 65;
+  const minWidth = (DEFAULT_MIN_WIDTH * DEFAULT_DPI) / DEFAULT_DPI;
+
+  let nominalCount = minWidth * pointResolution;
+  let suffix = "";
+
+  if (nominalCount < 0.001) {
+    suffix = "μm";
+    pointResolution *= 1000000;
+  } else if (nominalCount < 1) {
+    suffix = "mm";
+    pointResolution *= 1000;
+  } else if (nominalCount < 1000) {
+    suffix = "m";
+  } else {
+    suffix = "km";
+    pointResolution /= 1000;
+  }
 
-  return applyInverseTransform({ coordinate: point, affine: affine })
+  return suffix;
 }
 
+/**
+ * Get a num content item instance given value, code, meaning and units
+ *
+ * @param {number} value the measurement value
+ * @param {string} nameCodedConceptValue the coded concept value
+ * @param {string} nameCodedConceptMeaning the value coded concept meaning
+ * @param {string} unitCodedConceptValue the unit coded concept value
+ * @param {string} unitCodedConceptMeaning the unit coded concept meaning
+ * @returns {object} num content item instance
+ */
+const getMeasurementContentItem = (
+  value,
+  nameCodedConceptValue,
+  nameCodedConceptMeaning,
+  unitCodedConceptValue,
+  unitCodedConceptMeaning
+) => {
+  return new dcmjs.sr.valueTypes.NumContentItem({
+    name: new dcmjs.sr.coding.CodedConcept({
+      value: nameCodedConceptValue,
+      meaning: nameCodedConceptMeaning,
+      schemeDesignator: "DCM",
+    }),
+    value,
+    unit: new dcmjs.sr.coding.CodedConcept({
+      value: unitCodedConceptValue,
+      meaning: unitCodedConceptMeaning,
+      schemeDesignator: "DCM",
+    }),
```

Comment:
The `MeasurementUnitsSequence` must be a Sequence.

```suggestion
    unit: [new dcmjs.sr.coding.CodedConcept({
      value: unitCodedConceptValue,
      meaning: unitCodedConceptMeaning,
      schemeDesignator: "DCM",
    })],
```

I would actually consider this a bug in dcmjs (see [dcmjs.sr.valueTypes.NumContentItem](https://github.com/dcmjs-org/dcmjs/blob/678543c46df63ec9c11e1f5a97e1776242fea2a6/src/sr/valueTypes.js#L328)). However, I am not sure whether @pieper would be comfortable changing the type of the option and breaking backwards compatibility. Alternatively, we could allow both an array containing one item or an object representing an individual item.

Reply:
I really wanted to avoid having to make single-element sequences be Arrays.  My hope was that because the VR of SQ would be enough to handle any ambiguity.  As I recall though there are some corner cases so I think we did try to make the code consistently map SQ to Array.

Reply:
I just thought since its called `unit` rather than `units`, it may be more intuitive to pass an individual item as an object. However, I think we can keep the API as is and just improve documentation.

Reply:
@igoroctaviano this would still need to be addressed. Otherwise the generated SR document will not be standard compliant.